src/pages/index.js

```diff
@@ -1,101 +1,245 @@
+      userInfo.setUserInfo(userData);
+      editProfilePopup.close();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      editProfilePopup.renderLoading(false);
+    });
 }
 
-// BUTTONS////
+function handleAddCardSubmit({ title, url }) {
+  addCardPopup.renderLoading(true);
+
+  api
+    .addCard({ name: title, link: url })
+    .then((cardData) => {
+      const cardElement = createCard(cardData);
+      section.addItem(cardElement);
+      addCardPopup.close();
+      cardForm.reset();
+      addFormValidator.resetValidation();
+    })
+    .catch((err) => {
+      console.log(err);
+    })
+    .finally(() => {
+      addCardPopup.renderLoading(false);
+    });
+}
+
+const modalDelete = new PopupConfirmDelete("#modal__delete");
+modalDelete.setEventListeners();
+
+function handleDeleteSubmit(card) {
+  modalDelete.open();
+
+  modalDelete.setSubmit(() => {
+    modalDelete.renderLoading(true);
+    api
+      .deleteCard(card._id)
+      .then(() => {
+        card.deleteCardFromDom();
+        modalDelete.close();
+      })
+      .catch((err) => {
+        console.log(err);
+      })
+      .finally(() => {
+        modalDelete.renderLoading(false);
+      });
+  });
+}
+
+function handleLikeClicks(card) {
+  if (!card.isLiked) {
+    api
+      .likeCard(card._id)
+      .then(() => {
+        card.setIsLiked(true);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  } else {
+    api
+      .disLikeCard(card._id)
+      .then(() => {
+        card.setIsLiked(false);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+}
+
+api
+  .getUserInfo()
+  .then((data) => {
+    userInfo.updateAvatar(data);
+    userInfo.setUserInfo({
+      name: data.name,
+      about: data.about,
+    });
+  })
+  .catch((err) => {
+    console.log(err);
+  });
+
+/* -------------------------------------------------------------------------- */
+/*                                  Buttons                                   */
+/* -------------------------------------------------------------------------- */
 
 profileEditBtn.addEventListener("click", () => {
   const cardData = userInfo.getUserInfo();
-  modalInputTitle.value = cardData.title;
+  modalInputTitle.value = cardData.name;
   profileInputDescrption.value = cardData.description;
-  popupEditForm.open();
+  editFormValidator.resetValidation();
+  editProfilePopup.open();
 });
 
-addNewCardButton.addEventListener("click", () => popupAddForm.open());
+avatarEditBtn.addEventListener("click", () => editAvatarPopup.open());
+
+addNewCardButton.addEventListener("click", () => addCardPopup.open());
+
+/* -------------------------------------------------------------------------- */
+/*                                  Validations                                   */
+/* -------------------------------------------------------------------------- */
+const addFormValidator = new FormValidator(config, cardForm);
```

Comment:
**Could be improved**

You can **universally** create instances of validators for all forms in the project storing them inside one object: `formValidators`.  Then you can take any validator using attribute `name` of the form where you need to disable/enable the submit button or remove errors.

```js
// define an object for storing validators
const formValidators = {}

const enableValidation = (config) => {
  const formList = Array.from(document.querySelectorAll(config.formSelector))
  formList.forEach((formElement) => {
    const validator = new FormValidator(config, formElement)
    // Here you get the name of the form (if you don’t have it then you need to add it into each form in `index.html` first)
    const formName = formElement.getAttribute('name')

   // Here you store the validator using the `name` of the form
    formValidators[formName] = validator;
    validator.enableValidation();
  });
};

enableValidation(config);

```

And now you can use them for disabling buttons or clearing errors:

```js

formValidators[ profileForm.getAttribute('name') ].resetValidation()

// or you can use a string – the name of the form (you know it from `index.html`)

formValidators['profile-form'].resetValidation()

```